src/test/java/io/vertx/core/net/impl/SSLAllocatorTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.net.impl;
+
+import javax.net.ssl.SSLContext;
+
+import org.junit.Test;
+
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.handler.ssl.SslProvider;
+import io.vertx.core.buffer.impl.PartialPooledByteBufAllocator;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.impl.ContextInternal;
+import io.vertx.core.net.JdkSSLEngineOptions;
+import io.vertx.core.net.OpenSSLEngineOptions;
+import io.vertx.core.net.SSLOptions;
+import io.vertx.test.core.VertxTestBase;
+import io.vertx.test.tls.Cert;
+import io.vertx.test.tls.Trust;
+
+public class SSLAllocatorTest extends VertxTestBase {
```

Comment:
@cescoffier not yet perfect but decent enough to make sure the default behaviour is still the same
I need to wait till this is merged to add some CI run which enable both this and the unified allocator changes at https://github.com/eclipse-vertx/vert.x/pull/5292 , which is ready for being reviewed too (by @vietj )